### PR TITLE
Add Transform to KeyInfo and SignedSignatureProperties 

### PR DESCRIPTION
**Pull request by emmanvg**
This is the second PR related to issue #108 

It uses getCanonicalizationAlgorithmForSignature() from the algorithmProvider. Note: I built this PR with the changes from #110. Make sure you accept and resolve any conflicts there first.

Unfortunately I could not find a better way to check if the Algorithm is a C14N :(

### CODE REVIEW

File: src/main/java/xades4j/production/SignerBES.java [hunk 1]
```diff
@@ -244,14 +248,27 @@ public final XadesSignatureResult sign(
...
 
+            // Use same canonicalization URI as specified in the ds:CanonicalizationMethod for Signature.
+            String canonAlg = this.algorithmsProvider.getCanonicalizationAlgorithmForSignature().getUri();
+
             try
             {
-                signature.addDocument('#' + signedPropsId, null, digestAlgUri, null, QualifyingProperty.SIGNED_PROPS_TYPE_URI);
+                Transforms transforms = null;
+
+                if (!StringUtils.isNullOrEmptyString(canonAlg))
```
Comment (luisgoncalves) [inline src/main/java/xades4j/production/SignerBES.java]: `getCanonicalizationAlgorithmForSignature` is not supposed to return null, nor should its `getUri`. I think it wouldn't be a problem to always add the canonicalization algorithm.

File: src/main/java/xades4j/production/SignerBES.java [hunk 2]
```diff
@@ -244,14 +248,27 @@ public final XadesSignatureResult sign(
...
+                {
+                    // HACK: since we're not using Canonicalizer, do a quick check to ensure
+                    // that 'c14n' refers to a configured C14N algorithm.
+                    transforms = new Transforms(signatureDocument);
+                    transforms.addTransform(Canonicalizer.getInstance(canonAlg).getURI());
```
Comment (luisgoncalves) [inline src/main/java/xades4j/production/SignerBES.java]: Doesn't support possible parameters on the c14n algorithm.

File: src/main/java/xades4j/production/SignerBES.java [hunk 3]
```diff
@@ -260,6 +277,17 @@ public final XadesSignatureResult sign(
                 throw new UnsupportedAlgorithmException(
                         "Digest algorithm not supported in the XML Signature provider",
                         digestAlgUri, ex);
+            } catch (InvalidCanonicalizerException ex)
```
Comment (luisgoncalves) [inline src/main/java/xades4j/production/SignerBES.java]: Maybe all the usages of the HACK could be wrapped in a dedicated class...

File: src/main/java/xades4j/production/SignerBES.java (hunk 1 above)
Comment (emmanvg) [inline src/main/java/xades4j/production/SignerBES.java]: Hi Luis,

Thanks for the feedback! Based on your suggestion, I will remove the extra checks and always add the canonicalization algorithm. Please verify the commit https://github.com/luisgoncalves/xades4j/pull/111/commits/c6bd19eff6eea4238918bf84d4ecb64c2bd59762 . Thanks!

File: src/main/java/xades4j/production/SignerBES.java (hunk 3 above)
Comment (emmanvg) [inline src/main/java/xades4j/production/SignerBES.java]: Great suggestion. Based on your feedback, I re-introduced CanonicalizerUtils class. For now, it only has the "HACK" approach defined. I hope this was your interpretation. Note that, I do not perform a check for null. See commit https://github.com/luisgoncalves/xades4j/pull/111/commits/c6bd19eff6eea4238918bf84d4ecb64c2bd59762 and https://github.com/luisgoncalves/xades4j/pull/111/commits/c6bd19eff6eea4238918bf84d4ecb64c2bd59762#diff-d020145c5aef76d0f8394ec75d621dfa.

File: src/main/java/xades4j/production/SignerBES.java (hunk 2 above)
Comment (emmanvg) [inline src/main/java/xades4j/production/SignerBES.java]: Hi Luis,

Thanks for pointing this out! Based on your comments I made changes to the addTransform(..) method. Now, I was doubtful of what were those possible parameters and made some changes based on my interpretation. Can you please double check if I made the appropriate changes in: 

- https://github.com/luisgoncalves/xades4j/pull/111/commits/c6bd19eff6eea4238918bf84d4ecb64c2bd59762#diff-f8d676ba65712f301896cbc66c5bc987R85 
- https://github.com/luisgoncalves/xades4j/pull/111/commits/c6bd19eff6eea4238918bf84d4ecb64c2bd59762#diff-b5de685d575fcc090764640103b267d2R257

Let me know if this was your interpretation as well.

Comment (luisgoncalves) [inline src/main/java/xades4j/production/SignerBES.java]: I mean parameters of the algorithm itself, using AlgorithmParametersMarshaller. But I'll add this later. I've been picky enough :)

**Comment by luisgoncalves** [comment]
LGTM. I'll merge as soon as I have time.

**Pull request merged**